static/js/drafts.js

```diff
@@ -128,6 +128,19 @@ exports.restore_draft = function (draft_id) {
                               draft_copy);
     }
 
+    if (draft.type === "stream") {
+        if (draft.stream !== "") {
+            narrow.activate([{operator: "stream", operand: draft.stream},
+                             {operator: "topic", operand: draft.subject}],
+                             {select_first_unread: true, trigger: "restore draft"});
+        }
+    } else {
+        if (draft.private_message_recipient !== "") {
+            narrow.activate([{operator: "pm-with", operand: draft.private_message_recipient}],
+                             {select_first_unread: true, trigger: "restore draft"});
+        }
+    }
+
```

Comment:
If the restore-narrow experience with this code is as smooth as on the testing server, this looks all good to me! If there'll be a short delay, maybe we should consider only calling `narrow.activate` when we are not already narrowed to that topic, to improve the experience in such cases.